### PR TITLE
Manitoba child benefit

### LINKED ISSUE #429: Manitoba Child Benefits
Reference: 
https://www.gov.mb.ca/fs/eia/mcb.html
https://www.gov.mb.ca/fs/eia_manual/regs/4.html
https://www.gov.mb.ca/fs/eia/mcb_table.html

### PR DESCRIPTION
**Pull request by 7qiwx**
Fixes #429

### CODE REVIEW

**Comment by ZiqianCui** [comment]
Hi, remember to make format and git pull upstream master

File: policyengine_canada/parameters/gov/provinces/mb/benefits/mbcb/base.yaml [hunk 1]
```diff
@@ -0,0 +1,11 @@
+description: Manitoba provides this base amount for its Child Tax Benefit.
+values:
+  2023-01-01: 420 #35 per month
+metadata:
+  unit: currency-CAD
+  label: Manitoba child benefit base
+  reference:
+    - title: Manitoba child tax benefit 
+      href: https://www.gov.mb.ca/fs/eia/mcb.html
+    - title: Manitoba child benefit regulation
```
Comment (YiweiJ) [inline policyengine_canada/parameters/gov/provinces/mb/benefits/mbcb/base.yaml:10]:
```suggestion
    - title: Manitoba child benefit regulation § 6(1)
```

**Comment by YiweiJ** [comment]
Could try again run `git pull upstream master` after committed all changes

File: policyengine_canada/parameters/gov/provinces/mb/benefits/mbcb/base.yaml [hunk 2]
```diff
@@ -0,0 +1,11 @@
+description: Manitoba provides this base amount for its Child Tax Benefit.
+values:
+  2023-01-01: 420 #35 per month
```
Comment (mylittlebecca) [inline policyengine_canada/parameters/gov/provinces/mb/benefits/mbcb/base.yaml:3]: need to add 2021 and 2022 values? If Manitoba does not have that value in 2021 and 2022, 2021-01-01: 0

File: policyengine_canada/parameters/gov/provinces/mb/benefits/mbcb/eligible_age.yaml [hunk 3]
```diff
@@ -0,0 +1,9 @@
+description: Manitoba child benefit child age eligibility.
```
Comment (mylittlebecca) [inline policyengine_canada/parameters/gov/provinces/mb/benefits/mbcb/eligible_age.yaml:1]: Manitoba provides child benefit with children under this age.

File: policyengine_canada/parameters/gov/provinces/mb/benefits/mbcb/reduction/five_children.yaml [hunk 4]
```diff
@@ -0,0 +1,26 @@
+description: Manitoba Child Benefit phase out rate for families with five children.
```
Comment (mylittlebecca) [inline policyengine_canada/parameters/gov/provinces/mb/benefits/mbcb/reduction/five_children.yaml:1]: Manitoba phases the child benefit out for families with five children at this rate.

File: policyengine_canada/parameters/gov/provinces/mb/benefits/mbcb/reduction/four_children.yaml [hunk 5]
```diff
@@ -0,0 +1,25 @@
+description: Manitoba Child Benefit phase out rate for families with four children.
```
Comment (mylittlebecca) [inline policyengine_canada/parameters/gov/provinces/mb/benefits/mbcb/reduction/four_children.yaml:1]: Manitoba phases the child benefit out for families with four children at this rate.

File: policyengine_canada/parameters/gov/provinces/mb/benefits/mbcb/reduction/one_child.yaml [hunk 6]
```diff
@@ -0,0 +1,26 @@
+description: Manitoba Child Benefit phase out rate for families with one child.
```
Comment (mylittlebecca) [inline policyengine_canada/parameters/gov/provinces/mb/benefits/mbcb/reduction/one_child.yaml:1]: Manitoba phases the child benefit out for families with one child at this rate.

File: policyengine_canada/parameters/gov/provinces/mb/benefits/mbcb/reduction/six_or_more_children.yaml [hunk 7]
```diff
@@ -0,0 +1,28 @@
+description: Manitoba Child Benefit phase out rate for families with six children.
```
Comment (mylittlebecca) [inline policyengine_canada/parameters/gov/provinces/mb/benefits/mbcb/reduction/six_or_more_children.yaml:1]: Manitoba phases the child benefit out for families with six children at this rate.

File: policyengine_canada/parameters/gov/provinces/mb/benefits/mbcb/reduction/three_children.yaml [hunk 8]
```diff
@@ -0,0 +1,26 @@
+description: Manitoba Child Benefit phase out rate for families with three children.
```
Comment (mylittlebecca) [inline policyengine_canada/parameters/gov/provinces/mb/benefits/mbcb/reduction/three_children.yaml:1]: Manitoba phases the child benefit out for families with three children at this rate.

File: policyengine_canada/parameters/gov/provinces/mb/benefits/mbcb/reduction/two_children.yaml [hunk 9]
```diff
@@ -0,0 +1,26 @@
+description: Manitoba Child Benefit phase out rate for families with two children.
```
Comment (mylittlebecca) [inline policyengine_canada/parameters/gov/provinces/mb/benefits/mbcb/reduction/two_children.yaml:1]: Manitoba phases the child benefit out for families with two children at this rate.

**Comment by Xinyi-Shao** [comment]
You may run git pull upstream master to make sure the version is updated

File: policyengine_canada/variables/gov/provinces/mb/tax/income/benefits/child_benefit/mb_child_benefit_eligible_children.py [hunk 10]
```diff
@@ -0,0 +1,11 @@
+from policyengine_canada.model_api import *
+
+
+class mb_child_benefit_eligible_children(Variable):
+    value_type = int
+    entity = Household
+    label = "Manitoba child benefit eligible children"
```
Comment (Yaohhhh) [inline policyengine_canada/variables/gov/provinces/mb/tax/income/benefits/child_benefit/mb_child_benefit_eligible_children.py:7]: label = "Manitoba child benefit eligible children amount"

File: policyengine_canada/variables/gov/provinces/mb/tax/income/benefits/child_benefit/mb_child_benefit_eligible_children.py [hunk 11]
```diff
@@ -0,0 +1,11 @@
+from policyengine_canada.model_api import *
+
+
+class mb_child_benefit_eligible_children(Variable):
```
Comment (Yaohhhh) [inline policyengine_canada/variables/gov/provinces/mb/tax/income/benefits/child_benefit/mb_child_benefit_eligible_children.py:4]: mb_child_benefit_eligible_children_amount

File: policyengine_canada/variables/gov/provinces/mb/tax/income/benefits/child_benefit/mb_child_benefit_eligible_child.py [hunk 12]
```diff
@@ -0,0 +1,14 @@
+from policyengine_canada.model_api import *
+
+
+class mb_child_benefit_eligible_child(Variable):
```
Comment (Yaohhhh) [inline policyengine_canada/variables/gov/provinces/mb/tax/income/benefits/child_benefit/mb_child_benefit_eligible_child.py:4]: class mb_child_benefit_eligibility(Variable):

File: policyengine_canada/variables/gov/provinces/mb/tax/income/benefits/child_benefit/mb_child_benefit_eligible_child.py [hunk 13]
```diff
@@ -0,0 +1,14 @@
...
+    value_type = bool
+    entity = Person
+    label = "Manitoba child benefit eligible child"
```
Comment (Yaohhhh) [inline policyengine_canada/variables/gov/provinces/mb/tax/income/benefits/child_benefit/mb_child_benefit_eligible_child.py:7]: label = "Manitoba child benefit eligibility"

File: policyengine_canada/parameters/gov/provinces/mb/benefits/mbcb/base.yaml [hunk 14]
```diff
@@ -0,0 +1,14 @@
+description: Manitoba provides this base amount for its Child Tax Benefit.
```
Comment (PavelMakarchuk) [inline policyengine_canada/parameters/gov/provinces/mb/benefits/mbcb/base.yaml:1]:
```suggestion
description: Manitoba provides this base amount under the Child Tax Benefit.
```

File: policyengine_canada/parameters/gov/provinces/mb/benefits/mbcb/base.yaml [hunk 15]
```diff
@@ -0,0 +1,14 @@
...
+values:
+  2022-01-01: 420
+  2023-01-01: 420 #35 per month
```
Comment (PavelMakarchuk) [inline policyengine_canada/parameters/gov/provinces/mb/benefits/mbcb/base.yaml:4]:
```suggestion
  2023-01-01: 420 #35 per month
```

File: policyengine_canada/parameters/gov/provinces/mb/benefits/mbcb/base.yaml [hunk 16]
```diff
@@ -0,0 +1,14 @@
+description: Manitoba provides this base amount for its Child Tax Benefit.
+values:
+  2022-01-01: 420
```
Comment (PavelMakarchuk) [inline policyengine_canada/parameters/gov/provinces/mb/benefits/mbcb/base.yaml:3]:
```suggestion
  2022-01-01: 420 # 35 per month
```

File: policyengine_canada/parameters/gov/provinces/mb/benefits/mbcb/eligible_age.yaml [hunk 17]
```diff
@@ -0,0 +1,9 @@
+description: Manitoba provides child benefit with children under this age.
+values:
+  2023-01-01: 18
```
Comment (PavelMakarchuk) [inline policyengine_canada/parameters/gov/provinces/mb/benefits/mbcb/eligible_age.yaml:3]: need 2022 value

File: policyengine_canada/parameters/gov/provinces/mb/benefits/mbcb/reduction/five_children.yaml [hunk 18]
```diff
@@ -0,0 +1,26 @@
+description: Manitoba phases the child benefit out for families with five children at this rate.
+brackets:
+  - threshold:
+      2023-01-01: 0
```
Comment (PavelMakarchuk) [inline policyengine_canada/parameters/gov/provinces/mb/benefits/mbcb/reduction/five_children.yaml:4]: need 2022 values

Comment (PavelMakarchuk) [reply policyengine_canada/parameters/gov/provinces/mb/benefits/mbcb/reduction/five_children.yaml:4]: in all yaml files

File: policyengine_canada/parameters/gov/provinces/mb/benefits/mbcb/base.yaml [hunk 19]
```diff
@@ -0,0 +1,15 @@
+description: Manitoba provides this base amount under the Child Tax Benefit.
+values:
+  2022-01-01: 420 # 35 per month
+  2023-01-01: 420 # 35 per month
```
Comment (PavelMakarchuk) [inline policyengine_canada/parameters/gov/provinces/mb/benefits/mbcb/base.yaml:4]:
```suggestion
```

File: policyengine_canada/parameters/gov/provinces/mb/benefits/mbcb/eligible_age.yaml [hunk 20]
```diff
@@ -0,0 +1,10 @@
+description: Manitoba provides child benefit with children under this age.
+values:
+  2022-01-01: 18
+  2023-01-01: 18
```
Comment (PavelMakarchuk) [inline policyengine_canada/parameters/gov/provinces/mb/benefits/mbcb/eligible_age.yaml:4]:
```suggestion
```

File: policyengine_canada/parameters/gov/provinces/mb/benefits/mbcb/reduction/five_children.yaml [hunk 21]
```diff
@@ -0,0 +1,32 @@
+description: Manitoba phases the child benefit out for families with five children at this rate.
+brackets:
+  - threshold:
+      2022-01-01: 0
+      2023-01-01: 0
```
Comment (PavelMakarchuk) [inline policyengine_canada/parameters/gov/provinces/mb/benefits/mbcb/reduction/five_children.yaml:5]: if the values are exactly the same -> we use only the older date -> only use 2022-01-01:

Comment (PavelMakarchuk) [reply policyengine_canada/parameters/gov/provinces/mb/benefits/mbcb/reduction/five_children.yaml:5]: in all yaml files

File: policyengine_canada/variables/gov/provinces/mb/tax/income/benefits/child_benefit/mb_child_benefit.py [hunk 22]
```diff
@@ -0,0 +1,36 @@
...
+class mb_child_benefit(Variable):
+    value_type = float
+    entity = Household
+    label = "Manitoba Child Benefit"
+    definition_period = YEAR
+    defined_for = ProvinceCode.MB
+
+    def formula(household, period, parameters):
+        children = household("mb_child_benefit_eligible_children", period)
+        children = min_(children, 6)  # <= 6
```
Comment (PavelMakarchuk) [inline policyengine_canada/variables/gov/provinces/mb/tax/income/benefits/child_benefit/mb_child_benefit.py:13]: need to make the "6" a parameter `max_child_count`

File: policyengine_canada/parameters/gov/provinces/mb/benefits/mbcb/max_child_count.yaml [hunk 23]
```diff
@@ -0,0 +1,9 @@
+description: Manitoba provides child benefit with children under this count.
+values:
+  2022-01-01: 6
+metadata:
+  unit: year
+  label: Manitoba child benefit maximum child count
+  reference:
+    - title: Manitoba basic child age benefit
```
Comment (PavelMakarchuk) [inline policyengine_canada/parameters/gov/provinces/mb/benefits/mbcb/max_child_count.yaml:8]: I dont think the reference shows the "6"

File: policyengine_canada/parameters/gov/provinces/mb/benefits/mbcb/max_child_count.yaml [hunk 24]
```diff
@@ -0,0 +1,9 @@
+description: Manitoba provides child benefit with children under this count.
```
Comment (PavelMakarchuk) [inline policyengine_canada/parameters/gov/provinces/mb/benefits/mbcb/max_child_count.yaml:1]:
```suggestion
description: Manitoba adds an additional child benefit amount for up to this child count.
```

File: policyengine_canada/parameters/gov/provinces/mb/benefits/mbcb/max_child_count.yaml (hunk 23 above)
Comment (PavelMakarchuk) [reply policyengine_canada/parameters/gov/provinces/mb/benefits/mbcb/max_child_count.yaml:8]: use this: https://www.gov.mb.ca/fs/eia/mcb_table.html

File: policyengine_canada/parameters/gov/provinces/mb/benefits/mbcb/reduction/three_children.yaml [hunk 25]
```diff
@@ -0,0 +1,26 @@
...
+      
+metadata:
+  type: marginal_rate
+  threshold_unit: currency-CAD
+  rate_unit: /1
+  label: Manibota Child Benefit three children phase out rate
+  reference:
+    - title: Manibota child tax benefit 
+      href: https://www.gov.mb.ca/fs/eia/mcb_table.html
+    - title: Manibota child tax benefit regulation
```
Comment (PavelMakarchuk) [inline policyengine_canada/parameters/gov/provinces/mb/benefits/mbcb/reduction/three_children.yaml:24]: state specific parts of the source where the reduction percentage can be found - in all reduction yaml files

File: changelog_entry.yaml [hunk 26]
```diff
@@ -0,0 +1,4 @@
+- bump: minor
+  changes:
+    added:
+    - Manitoba Child Benefit.
```
Comment (PavelMakarchuk) [inline changelog_entry.yaml:4]:
```suggestion
    - Manitoba child benefit.
```

File: policyengine_canada/parameters/gov/provinces/mb/benefits/mbcb/reduction/six_or_more_children.yaml [hunk 27]
```diff
@@ -0,0 +1,28 @@
...
+
+metadata:
+  type: marginal_rate
+  threshold_unit: currency-CAD
+  rate_unit: /1
+  label: Manibota Child Benefit one child phase out rate
+  reference:
+    - title: Manibota child tax benefit 
+      href: https://www.gov.mb.ca/fs/eia/mcb_table.html
+      # Law does specify rate for 7+ children, assume that it is the same as for 6
```
Comment (PavelMakarchuk) [inline policyengine_canada/parameters/gov/provinces/mb/benefits/mbcb/reduction/six_or_more_children.yaml:26]:
```suggestion
      # Law does specify reduction start for 7+ children, assume that it is the same as for 6
```

File: policyengine_canada/parameters/gov/provinces/mb/benefits/mbcb/eligible_age.yaml [hunk 28]
```diff
@@ -0,0 +1,9 @@
+description: Manitoba provides child benefit with children under this age.
```
Comment (PavelMakarchuk) [inline policyengine_canada/parameters/gov/provinces/mb/benefits/mbcb/eligible_age.yaml:1]:
```suggestion
description: Manitoba qualifies children for the child tax benefit below this age.
```

File: policyengine_canada/parameters/gov/provinces/mb/benefits/mbcb/reduction/four_children.yaml [hunk 29]
```diff
@@ -0,0 +1,26 @@
...
+  - threshold:
+      2022-01-01: 22_242
+    rate:
+      2022-01-01: 1 
+
+metadata:
+  type: marginal_rate
+  threshold_unit: currency-CAD
+  rate_unit: /1
+  label: Manibota Child Benefit one child phase out rate
```
Comment (PavelMakarchuk) [inline policyengine_canada/parameters/gov/provinces/mb/benefits/mbcb/reduction/four_children.yaml:21]:
```suggestion
  label: Manibota Child Benefit four children phase out rate
```

File: policyengine_canada/parameters/gov/provinces/mb/benefits/mbcb/reduction/six_or_more_children.yaml (hunk 27 above)
Comment (MaxGhenis) [reply policyengine_canada/parameters/gov/provinces/mb/benefits/mbcb/reduction/six_or_more_children.yaml]: Does or does not?

File: policyengine_canada/parameters/gov/provinces/mb/benefits/mbcb/reduction/six_or_more_children.yaml [hunk 30]
```diff
@@ -0,0 +1,26 @@
...
+      
+metadata:
+  type: marginal_rate
+  threshold_unit: currency-CAD
+  rate_unit: /1
+  label: Manitoba Child Benefit six or more children phase out rate
+  reference:
+    - title: Manitoba child tax benefit 
+      href: https://www.gov.mb.ca/fs/eia/mcb_table.html
+      # Law does specify reduction start for 7+ children, assume that it is the same as for 6
```
Comment (PavelMakarchuk) [inline policyengine_canada/parameters/gov/provinces/mb/benefits/mbcb/reduction/six_or_more_children.yaml:24]:
```suggestion
      # Law does not specify the reduction start for 7+ children, assuming that it is the same as for 6
```

File: policyengine_canada/parameters/gov/provinces/mb/benefits/mbcb/reduction/six_or_more_children.yaml (hunk 27 above)
Comment (PavelMakarchuk) [reply policyengine_canada/parameters/gov/provinces/mb/benefits/mbcb/reduction/six_or_more_children.yaml]: does not*

File: policyengine_canada/parameters/gov/provinces/mb/benefits/mbcb/base.yaml
Comment (PavelMakarchuk) [inline policyengine_canada/parameters/gov/provinces/mb/benefits/mbcb/base.yaml:1]: add to the list of credits (I think - need to check where in the net income tree this belongs)

File: policyengine_canada/parameters/gov/provinces/mb/benefits/mbcb/eligible_age.yaml [hunk 31]
```diff
@@ -0,0 +1,9 @@
+description: Manitoba qualifies children for the child tax benefit below this age.
+values:
+  2022-01-01: 18
+metadata:
+  unit: year
```
[the remark this answers is not in the document]
Comment (PavelMakarchuk) [reply policyengine_canada/parameters/gov/provinces/mb/benefits/mbcb/eligible_age.yaml:5]: and other yaml files

File: policyengine_canada/parameters/gov/provinces/mb/benefits/mbcb/eligible_age.yaml [hunk 32]
```diff
@@ -0,0 +1,9 @@
...
+  label: Manitoba child benefit age benefit eligibility
```
Comment (PavelMakarchuk) [inline policyengine_canada/parameters/gov/provinces/mb/benefits/mbcb/eligible_age.yaml:6]:
```suggestion
  label: Manitoba child benefit age eligibility
```

File: policyengine_canada/parameters/gov/provinces/mb/benefits/mbcb/eligible_age.yaml [hunk 33]
```diff
@@ -0,0 +1,9 @@
...
+  reference:
+    - title: Manitoba basic child age benefit
```
Comment (PavelMakarchuk) [inline policyengine_canada/parameters/gov/provinces/mb/benefits/mbcb/eligible_age.yaml:8]:
```suggestion
    - title: Province of Manitoba - Manitoba Child Benefit

```

File: policyengine_canada/parameters/gov/provinces/mb/benefits/mbcb/max_child_count.yaml [hunk 34]
```diff
@@ -0,0 +1,9 @@
+description: Manitoba adds an additional child benefit amount for up to this child count.
+values:
+  2022-01-01: 6
+metadata:
+  unit: year
+  label: Manitoba child benefit maximum child count
+  reference:
+    - title: Manitoba basic child age benefit
+      href: https://www.gov.mb.ca/fs/eia/mcb_table.html
```
Comment (PavelMakarchuk) [inline policyengine_canada/parameters/gov/provinces/mb/benefits/mbcb/max_child_count.yaml:9]: add law

**Review by PavelMakarchuk — CHANGES_REQUESTED** [review]

File: policyengine_canada/parameters/gov/provinces/mb/benefits/mbcb/base.yaml [hunk 35]
```diff
@@ -0,0 +1,15 @@
...
+  unit: currency-CAD
+  label: Manitoba child benefit base
+  period: year
+  reference:
+    - title: Manitoba child tax benefit 
+      href: https://www.gov.mb.ca/fs/eia/mcb.html
+    - title: Manitoba child benefit regulation § 6(1)
+      href: https://www.gov.mb.ca/fs/eia_manual/regs/4.html
+    - title: Manitoba child benefit regulation § 16
+      href: https://www.gov.mb.ca/fs/eia_manual/regs/4.html
```
Comment (PavelMakarchuk) [inline policyengine_canada/parameters/gov/provinces/mb/benefits/mbcb/base.yaml:15]:
```suggestion
```

Comment (PavelMakarchuk) [reply policyengine_canada/parameters/gov/provinces/mb/benefits/mbcb/base.yaml:15]: dont need this 

- is there a tax form for 2022 that we can reference? If not please leave a comment

Same for all yaml

File: policyengine_canada/parameters/gov/provinces/mb/benefits/mbcb/max_child_count.yaml [hunk 36]
```diff
@@ -0,0 +1,12 @@
+description: Manitoba adds an additional child benefit amount for up to this child count.
```
Comment (PavelMakarchuk) [inline policyengine_canada/parameters/gov/provinces/mb/benefits/mbcb/max_child_count.yaml:1]:
```suggestion
description: Manitoba adds an additional child benefit amount for up to this children count.
```

File: policyengine_canada/parameters/gov/provinces/mb/benefits/mbcb/reduction/five_children.yaml [hunk 37]
```diff
@@ -0,0 +1,27 @@
+description: Manitoba phases the child benefit out for families with five children at this rate.
```
Comment (PavelMakarchuk) [inline policyengine_canada/parameters/gov/provinces/mb/benefits/mbcb/reduction/five_children.yaml:1]:
```suggestion
description: Manitoba phases the child benefit out for families with five children at this rate, based on income.
```

Comment (PavelMakarchuk) [reply policyengine_canada/parameters/gov/provinces/mb/benefits/mbcb/reduction/five_children.yaml:1]: all reduction yaml files

File: policyengine_canada/parameters/gov/provinces/mb/tax/income/credits/refundable.yaml [hunk 38]
```diff
@@ -0,0 +1,17 @@
...
+  unit: list  
+  period: year 
+  label: Manitoba refundable credits  
+  reference:
+  - title: Manitoba child tax benefit 
+    href: https://www.gov.mb.ca/fs/eia/mcb.html
+  - title: Manitoba child benefit regulation § 6(1)
+    href: https://www.gov.mb.ca/fs/eia_manual/regs/4.html
+  - title: Manitoba child benefit regulation § 16
+    href: https://www.gov.mb.ca/fs/eia_manual/regs/4.html
```
Comment (PavelMakarchuk) [inline policyengine_canada/parameters/gov/provinces/mb/tax/income/credits/refundable.yaml:16]:
```suggestion
  # Child benefit reference
  - title: Manitoba child benefit regulation § 6(1)
    href: https://www.gov.mb.ca/fs/eia_manual/regs/4.html
```

**Review by PavelMakarchuk — CHANGES_REQUESTED** [review]

File: policyengine_canada/parameters/gov/provinces/mb/benefits/mbcb/base.yaml [hunk 39]
```diff
@@ -0,0 +1,14 @@
...
+metadata:
+  unit: currency-CAD
+  label: Manitoba child benefit base
+  period: year
+  reference:
+    - title: Manitoba child tax benefit 
+      href: https://www.gov.mb.ca/fs/eia/mcb.html
+    - title: Manitoba child benefit regulation § 6(1)
+      href: https://www.gov.mb.ca/fs/eia_manual/regs/4.html 
+    # did not find tax form for 2022
```
Comment (PavelMakarchuk) [inline policyengine_canada/parameters/gov/provinces/mb/benefits/mbcb/base.yaml:14]:
```suggestion
```

Comment (PavelMakarchuk) [reply policyengine_canada/parameters/gov/provinces/mb/benefits/mbcb/base.yaml:14]: remove everywhere

File: policyengine_canada/parameters/gov/provinces/mb/tax/income/credits/refundable.yaml
Comment (PavelMakarchuk) [inline policyengine_canada/parameters/gov/provinces/mb/tax/income/credits/refundable.yaml:1]: I am not sure if this is a non refundable credit - if it is not mentioned on a tax form then it is a benefit which is added separately, if you could check and if that is true, we need a benefits.py (not mb specific) where we add those

File: policyengine_canada/variables/gov/provinces/mb/tax/income/benefits/child_benefit/mb_child_benefit.py [hunk 40]
```diff
@@ -0,0 +1,37 @@
...
+    value_type = float
+    entity = Household
+    label = "Manitoba Child Benefit"
+    definition_period = YEAR
+    defined_for = ProvinceCode.MB
+
+    def formula(household, period, parameters):
+        children = household("mb_child_benefit_eligible_children", period)
+        p = parameters(period).gov.provinces.mb.benefits.mbcb
+        children = min_(children, p.max_child_count)  # <= 6
```
Comment (PavelMakarchuk) [inline policyengine_canada/variables/gov/provinces/mb/tax/income/benefits/child_benefit/mb_child_benefit.py:14]:
```suggestion
        capped_children_count = min_(children, p.max_child_count)  # <= 6
```

File: policyengine_canada/variables/gov/provinces/mb/tax/income/benefits/child_benefit/mb_child_benefit.py [hunk 41]
```diff
@@ -0,0 +1,37 @@
...
+
+class mb_child_benefit(Variable):
+    value_type = float
+    entity = Household
+    label = "Manitoba Child Benefit"
+    definition_period = YEAR
+    defined_for = ProvinceCode.MB
+
+    def formula(household, period, parameters):
+        children = household("mb_child_benefit_eligible_children", period)
```
Comment (PavelMakarchuk) [inline policyengine_canada/variables/gov/provinces/mb/tax/income/benefits/child_benefit/mb_child_benefit.py:12]:
```suggestion
        eligible_children = household("mb_child_benefit_eligible_children", period)
```

File: policyengine_canada/variables/gov/provinces/mb/tax/income/benefits/child_benefit/mb_child_benefit.py [hunk 42]
```diff
@@ -0,0 +1,37 @@
...
+        p = parameters(period).gov.provinces.mb.benefits.mbcb
+        children = min_(children, p.max_child_count)  # <= 6
+
+        income = household("adjusted_family_net_income", period)
+        base = children * p.base
```
Comment (PavelMakarchuk) [inline policyengine_canada/variables/gov/provinces/mb/tax/income/benefits/child_benefit/mb_child_benefit.py:17]:
```suggestion
        base_amount = children * p.base
```

**Review by PavelMakarchuk — CHANGES_REQUESTED** [review]

File: policyengine_canada/variables/household/income/household/household_benefits.py
Comment (PavelMakarchuk) [inline policyengine_canada/variables/household/income/household/household_benefits.py:1]: could you add the general child benefit and other federal benefits to this list

File: policyengine_canada/variables/household/income/household/household_benefits.py [hunk 43]
```diff
@@ -0,0 +1,17 @@
...
+
+
+class household_benefits(Variable):
+    value_type = float
+    entity = Household
+    label = "benefits"
+    unit = CAD
+    definition_period = YEAR
+    adds = [
+        "Canada_child_benefit",
```
Comment (PavelMakarchuk) [inline policyengine_canada/variables/household/income/household/household_benefits.py:11]: is that what the variables are called in the repo?

Comment (7qiwx) [reply policyengine_canada/variables/household/income/household/household_benefits.py:11]: the file is named "ccb"

Comment (7qiwx) [reply policyengine_canada/variables/household/income/household/household_benefits.py:11]: > is that what the variables are called in the repo?

the file is named "ccb"

Comment (PavelMakarchuk) [reply policyengine_canada/variables/household/income/household/household_benefits.py:11]: I think the variable is just called 
`child_benefit`
`dental_benefit`

File: policyengine_canada/parameters/gov/provinces/mb/benefits/mbcb/max_child_count.yaml [hunk 44]
```diff
@@ -0,0 +1,13 @@
...
+metadata:
+  unit: year
+  label: Manitoba child benefit maximum child count
+  period: year
+  reference:
+    - title: Manitoba basic child age benefit
+      href: https://www.gov.mb.ca/fs/eia/mcb_table.html
+    - title: Manitoba child benefit regulation § 5
+      href: https://www.gov.mb.ca/fs/eia_manual/regs/4.html 
+    # did not find tax form for 2022
```
Comment (PavelMakarchuk) [inline policyengine_canada/parameters/gov/provinces/mb/benefits/mbcb/max_child_count.yaml:13]: delete

File: policyengine_canada/parameters/gov/provinces/mb/benefits/mbcb/reduction/five_children.yaml [hunk 45]
```diff
@@ -0,0 +1,28 @@
...
+  rate_unit: /1
+  label: Manitoba Child Benefit five children phase out rate
+  period: year
+  reference:
+    - title: Manitoba child tax benefit 
+      href: https://www.gov.mb.ca/fs/eia/mcb_table.html
+    - title: Manitoba child tax benefit regulation 6(2)
+      href: https://www.gov.mb.ca/fs/eia_manual/regs/4.html
+
+
```
Comment (PavelMakarchuk) [inline policyengine_canada/parameters/gov/provinces/mb/benefits/mbcb/reduction/five_children.yaml:28]: delete extra line

File: policyengine_canada/variables/gov/provinces/mb/tax/income/benefits/child_benefit/mb_child_benefit.py [hunk 46]
```diff
@@ -0,0 +1,41 @@
...
+    defined_for = ProvinceCode.MB
+
+    def formula(household, period, parameters):
+        eligible_children = household(
+            "mb_child_benefit_eligible_children", period
+        )
+        p = parameters(period).gov.provinces.mb.benefits.mbcb
+        capped_children_count = min_(
+            eligible_children, p.max_child_count
+        )  # <= 6
```
Comment (PavelMakarchuk) [inline policyengine_canada/variables/gov/provinces/mb/tax/income/benefits/child_benefit/mb_child_benefit.py:18]:
```suggestion
        )  
        # <= 6
```

File: policyengine_canada/variables/household/income/household/household_benefits.py [hunk 47]
```diff
@@ -0,0 +1,17 @@
...
+class household_benefits(Variable):
+    value_type = float
+    entity = Household
+    label = "benefits"
+    unit = CAD
+    definition_period = YEAR
+    adds = [
+        "child_benefit",
+        "dental_benfit",
+        "mb_child_benefit",
```
Comment (PavelMakarchuk) [inline policyengine_canada/variables/household/income/household/household_benefits.py:13]:
```suggestion
        # Manitoba
        "mb_child_benefit",
```

Comment (PavelMakarchuk) [reply policyengine_canada/variables/household/income/household/household_benefits.py:13]: add all states

File: policyengine_canada/parameters/gov/provinces/mb/tax/income/credits/refundable.yaml [hunk 48]
```diff
@@ -0,0 +1,12 @@
...
+  2022-01-01: []
+    
+
+metadata:
+  unit: list  
+  period: year 
+  label: Manitoba refundable credits  
+  
+  
+
```
Comment (PavelMakarchuk) [inline policyengine_canada/parameters/gov/provinces/mb/tax/income/credits/refundable.yaml:12]: delete extra lines

**Review by PavelMakarchuk — CHANGES_REQUESTED** [review]

**Review by PavelMakarchuk — APPROVED** [review]

File: policyengine_canada/variables/household/income/household/household_benefits.py [hunk 49]
```diff
@@ -0,0 +1,21 @@
...
+
+class household_benefits(Variable):
+    value_type = float
+    entity = Household
+    label = "benefits"
+    unit = CAD
+    definition_period = YEAR
+    adds = [
+        "child_benefit",
+        "dental_benfit",
```
Comment (MaxGhenis) [inline policyengine_canada/variables/household/income/household/household_benefits.py:12]:
```suggestion
        "dental_benefit",
```

**Review by MaxGhenis — APPROVED** [review]

**Pull request merged**